senator/game.py

```diff
@@ -78,45 +97,93 @@ def nash_conv(self) -> float:
+            nash_convexity += np.max(instantaneous_regret)
 
         return nash_convexity
 
 
 class Game:
-    def __init__(self, initial_utility: np.ndarray, votes: np.ndarray):
-        assert len(initial_utility.shape) == 2, "Initial utility should be a 2D array"
+    def __init__(
+        self,
+        initial_utility: np.ndarray,
+        votes: np.ndarray,
+        is_owner: np.ndarray,
+        owner_loss: float = 1,
+        other_loss: float = 0,
+        owner_trashold: float = 0,
+    ) -> None:
+
+        self.owner_loss = owner_loss
+        self.other_loss = other_loss
+        self.owner_trashold = owner_trashold
+        self.is_owner = is_owner
         self.votes = votes
         self.initial_utility = initial_utility
-        root_utility_matrix = get_utility_matrix(initial_utility, votes)  # TODO: see what it takes to get the utility matrix
+        root_utility_matrix = get_utility_matrix(initial_utility, votes, is_owner)
         self.root = Node(parent=None, utility_matrix=root_utility_matrix)
 
-    def add_child(self, node: Node, joint_action: np.ndarray, current_utility: np.ndarray, votes: np.ndarray) -> None:
-        child_utility_matrix = get_utility_matrix(current_utility, votes)  # TODO: see what it takes to get the utility matrix
-        node.children[tuple(joint_action)] = Node(parent=node, utility_matrix=child_utility_matrix)
-
-    def solve_node_via_rm_plus(self, node: Node, iterations: int = 1000) -> None:
+    def add_child(
+        self,
+        node: Node,
+        joint_action: np.ndarray,
+        current_utility: np.ndarray,
+        votes: np.ndarray,
+    ) -> None:
+        child_utility_matrix = get_utility_matrix(
+            current_utility,
+            votes,
+            self.is_owner,
+            self.owner_loss,
+            self.other_loss,
+            self.owner_trashold,
+        )  # TODO: see what it takes to get the utility matrix
+        node.children[tuple(joint_action)] = Node(
+            parent=node, utility_matrix=child_utility_matrix
+        )
+
+    def solve_node_via_rm_plus(self, node: Node, iterations: int = 2000) -> None:
         for _ in range(iterations):
             node.rm_plus_step()
 
         if (conv := node.nash_conv()) >= 0.001:
             print(f"Nash Convexity is {conv}, something is wrong...")
 
-    def run_greedy_search(self, num_expansions: int = 10):
-        """
-        Run a greedy search algorithm, periodically expanding the tree. Each expansion is done by first solving the
-        node using the RM+ algorithm, sampling a joing action, and adding the child to the tree.
-        TODO: decide how to do the expanding. Baseline is to just do a path of length 5.
-        """
+    def solve_node_via_rm(self, node: Node, iterations: int = 2000) -> None:
+        for _ in range(iterations):
+            node.rm_step()
+
+        if (conv := node.nash_conv()) >= 0.001:
+            print(f"Nash Convexity is {conv}, something is wrong...")
```

Comment:
Možná by stálo za to udělat z tohohle svou vlastní metodu `self.check_nash_conv()`?

Reply:
Ale jestli se ti do toho nechce tak tohle resolvni a kašli na to xD